scipy/sparse/_base.py

```diff
@@ -67,7 +67,15 @@ class _spbase:
 
     __array_priority__ = 10.1
     _format = 'und'  # undefined
-    ndim = 2
+    
+    @property
+    def ndim(self) -> int:
+        return len(self._shape)
+
+    @property
+    def _shape_as_2d(self):
```

Comment:
Name is OK for now, but needs to be changed if we ever support any other dimensionality than 1 and 2.